Cotecna.Domain.Core/ApplicationMediator.cs

```diff
@@ -2,77 +2,73 @@
 
 namespace Cotecna.Domain.Core
 {
-    public interface ApplicationMediator
+    /// <summary>
+    /// Defines a mediator to encapsulate command and querys.
+    /// Represents <see cref="IApplicationMediator"/> Injectable Service Interface
+    /// </summary>
+    public interface IApplicationMediator
     {
 
         /// <summary>
-        /// Defines a mediator to encapsulate command and querys.
-        /// Represents <see cref="IApplicationMediator"/> Injectable Service Interface
+        /// Manage comand objects
         /// </summary>
-        public interface IApplicationMediator
-        {
+        /// <typeparam name="ICommand"> <see cref="ICommand"/></typeparam>
+        /// <param name="command">Command object to be dispatched</param>
+        /// <returns>Returns boolean if Command was correctly dispatched</returns>
+        bool Dispatch(ICommand command);
+
+        /// <summary>
+        /// Manage comand objects
+        /// </summary>
+        /// <typeparam name="IQuery"> <see cref="IQuery"/></typeparam>
+        /// <param name="query">Query object to be dispatched</param>
+        /// <returns>Result object</returns>
+        T Dispatch<T>(IQuery<T> query);
+    }
 
-            /// <summary>
-            /// Manage comand objects
-            /// </summary>
-            /// <typeparam name="ICommand"> <see cref="ICommand"/></typeparam>
-            /// <param name="command">Command object to be dispatched</param>
-            /// <returns>Returns boolean if Command was correctly dispatched</returns>
-            bool Dispatch(ICommand command);
-
-            /// <summary>
-            /// Manage comand objects
-            /// </summary>
-            /// <typeparam name="IQuery"> <see cref="IQuery"/></typeparam>
-            /// <param name="query">Query object to be dispatched</param>
-            /// <returns>Result object</returns>
-            T Dispatch<T>(IQuery<T> query);
-        }
 
 
+    /// <summary>
+    /// Defines a instance of IApplicationMediator <see cref="IApplicationMediator"/> 
+    /// </summary>
+    public sealed class ApplicationMediator : IApplicationMediator
```

Comment:
I'm really a fan of having the inferface in the same file in the case that the interface extraction is only because of DI purposes, like it's the case since it simplifies a lot the code comprehenssion and structure simplicity.

However guys if you think we have to move it I can do it, what do you think?

Reply:
I fully understand both sides. Difficult decision in that case. Just to say one option, since I also use to put the interface in the same implementation file were it's really small and only has one implementation, I vote for keeping it on the same file.